### PR TITLE
fix bugs issues 267

### PR DESCRIPTION
**Pull request by xmmwc**
fix message back from gateway which sid(MAC) lost first or more zero

### CODE REVIEW

**Comment by xmmwc** [comment]
#267

**Comment by liupeinye** [comment]
need this pr @YinHangCode

**Comment by liupeinye** [comment]
bro, your codes cannot work well. I still get that warn @xmmwc

File: index.js [hunk 1]
```diff
@@ -391,7 +391,7 @@ MiAqaraPlatform.prototype.initServerMiAqaraManage = function() {
 MiAqaraPlatform.prototype.sendMQTTMessage4ParseMessage = function(msg, rinfo) {
```
Comment (liupeinye) [inline index.js]: should modify this method: MiAqaraPlatform.prototype.parseMessage

Comment (liupeinye) [reply index.js]: not MQTT

File: index.js [hunk 2]
```diff
@@ -423,7 +443,7 @@ MiAqaraPlatform.prototype.parseMessage = function(msg, rinfo) {
     // check message
     var jsonObj;
     try {
-        jsonObj = JSON.parse(msg);
+        jsonObj = that.getJsonObj(msg);
```
Comment (liupeinye) [inline index.js]: sorry, you are right. Here goes the code

Comment (xmmwc) [reply index.js]: did my code solve your problem?it works for me.

Comment (xmmwc) [reply index.js]: looks like you are chinese so :  这代码我这跑着OK。但没试过其他环境和MQTT的需求。你也可以测试下

Comment (liupeinye) [reply index.js]: 兄弟，没毛病。完美解决问题，MQTT我也没需求，没法试。

**Comment by banzalik** [comment]
It is don't work for me too.

**Comment by xmmwc** [comment]
> It is don't work for me too.

Any logs?

**Comment by banzalik** [comment]
> > It is don't work for me too.
> 
> Any logs?
I am found problem.
New hubs don't support developer mode (has closed local port and user can't open this port by MI application)

manual how to fix it:
https://community.openhab.org/t/solved-openhab2-xiaomi-mi-gateway-does-not-respond/52963/124
(need soldering and UART to USB cable)

**Pull request closed without merging**